core/src/main/java/uk/gov/gchq/koryphe/iterable/ChainedIterator.java

```diff
@@ -63,12 +65,16 @@ private Iterator<? extends T> getIterator() {
         while (!currentIterator.hasNext()) {
             CloseableUtil.close(currentIterator);
             if (iterablesIterator.hasNext()) {
-                currentIterator = iterablesIterator.next().iterator();
+                Object next = iterablesIterator.next();
+                if (nonNull(next) && next instanceof Iterable) {
```

Comment:
null check not needed as null is not `instanceof Iterable`

Reply:
```suggestion
                if (next instanceof Iterable) {
```